mmv1/third_party/terraform/services/logging/resource_logging_project_bucket_config.go

```diff
@@ -260,6 +265,9 @@ func resourceLoggingProjectBucketConfigRead(d *schema.ResourceData, meta interfa
 	if err := d.Set("description", res["description"]); err != nil {
 		return fmt.Errorf("Error setting description: %s", err)
 	}
+	if err := d.Set("locked", res["locked"]); err != nil {
```

Comment:
Out of curiosity, why do we set these values even if they are the zero value? Don't we expect to use the !tpgresource.IsEmptyValue etc?

Reply:
I'm not sure I understand the question? If we don't set a top-level field in Read it doesn't take on the zero value- it preserves the old value recorded in state. That means we'd never detect drift from a true to false value. For a field like this that can only happen if the resource was recreated out of band, but in other cases this can matter more.

Reply:
I somehow read this as the Create method 🤦